app/views/comments/_comment_proper.html.erb

```diff
@@ -25,40 +25,55 @@
+            <span class="comment-username">
+              <span class="comment-username-inner">
+                <%= comment.user.name %>
+              </span>
             </span>
-          </span>
-        </a>
-        <% if comment.user.twitter_username.present? %>
-          <a href="https://twitter.com/<%= comment.user.twitter_username %>" rel="noopener noreferrer" target="_blank">
-            <%= image_tag("twitter-logo.svg", class: "icon-img", alt: "twitter logo") %></a>
-        <% end %>
-        <% if comment.user.github_username.present? %>
-          <a href="https://github.com/<%= comment.user.github_username %>" rel="noopener noreferrer" target="_blank">
-            <%= image_tag("github-logo.svg", class: "icon-img", alt: "github logo") %>
           </a>
+          <% if comment.user.twitter_username.present? %>
+            <a href="https://twitter.com/<%= comment.user.twitter_username %>" rel="noopener noreferrer" target="_blank">
+              <%= image_tag("twitter-logo.svg", class: "icon-img", alt: "twitter logo") %></a>
+          <% end %>
+          <% if comment.user.github_username.present? %>
+            <a href="https://github.com/<%= comment.user.github_username %>" rel="noopener noreferrer" target="_blank">
+              <%= image_tag("github-logo.svg", class: "icon-img", alt: "github logo") %>
+            </a>
+          <% end %>
+          <% if commentable_author_is_op?(commentable, comment) %>
+            <span class="op-marker"><%= get_ama_or_op_banner(commentable) %></span>
+          <% end %>
+          <%= render "comments/comment_date", decorated_comment: decorated_comment %>
         <% end %>
-        <% if commentable_author_is_op?(commentable, comment) %>
-          <span class="op-marker"><%= get_ama_or_op_banner(commentable) %></span>
-        <% end %>
-        <%= render "comments/comment_date", decorated_comment: decorated_comment %>
         <button class="dropbtn" aria-label="Toggle dropdown menu">
-          <%= image_tag("three-dots.svg", class: "dropdown-icon", alt: "Dropdown menu icon") %>
+          <%= image_tag("three-dots.svg", class: "dropdown-icon", alt: "Dropdown menu") %>
         </button>
         <div class="dropdown">
           <div class="dropdown-content">
-            <a href="<%= comment.path %>">
-              Permalink
-            </a>
+            <% unless should_be_hidden?(comment, @root_comment) %>
+              <a href="<%= comment.path %>">
+                Permalink
+              </a>
+            <% end %>
             <span class="comment-actions hidden" data-user-id="<%= comment.user_id %>" style="display: none;">
               <a href="<%= comment.path %>/settings" rel="nofollow" style="color:#0a0a0a;" data-no-instant>Settings</a>
             </span>
+            <% action = comment.hidden_by_commentable_user ? "Unhide" : "Hide" %>
```

Comment:
Even as not-the-author I'm able to see a dropdown with the hide option on a hidden comment.

<img width="819" alt="Screen Shot 2019-12-02 at 5 18 35 PM" src="https://user-images.githubusercontent.com/3102842/69999991-fe258100-1527-11ea-9452-0973cc17d51b.png">